src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/MessagePackHubProtocol.cs

```diff
@@ -20,10 +20,18 @@ public class MessagePackHubProtocol : IHubProtocol
         private const int VoidResult = 2;
         private const int NonVoidResult = 3;
 
+        private static readonly SerializationContext _serializationContext;
```

Comment:
Is this like Json formatting options or is it just for our part of the protocol?

Reply:
This is part of the protocol but we will probably need Json like formatting on top of that. This setting just says to write key/property value pairs instead of just values.

Reply:
Does this also make C# less efficient?

Reply:
The payload is bigger but without it is impossible to deserialize the payload if you don't have a strong type that can be reflected on. Note that this is only for custom/structural objects in arguments and result types.